core: Build fails on Visual Studio 2012

<!--
If you have a question rather than reporting a bug please go to http://answers.opencv.org where you get much faster responses.
If you need further assistance please read [How To Contribute](https://github.com/opencv/opencv/wiki/How_to_contribute).

This is a template helping you to create an issue which can be processed as quickly as possible. This is the bug reporting section for the OpenCV library.
-->

##### System information (version)
<!-- Example
- OpenCV => 3.1
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2015
-->

- OpenCV => master ( dcb3c4ff1e70c7301a985cbf62da04d946b0a72b )
- Operating System / Platform => Win 7 Pro 64bit
- Compiler => Visual Studio 2012

##### Detailed description
 - When building master on Visual Studio 2012, build fails
```
Error	1	error C1083: Cannot open include file: 'initializer_list': No such file or directory	modules\core\include\opencv2\core\matx.hpp	57	opencv_core
```
- The direct cause is [this include pragma here](https://github.com/opencv/opencv/blob/dcb3c4ff1e70c7301a985cbf62da04d946b0a72b/modules/core/include/opencv2/core/matx.hpp#L57)
- [Visual Studio 2012 doesn't support ```initializer_list``` ](https://blogs.msdn.microsoft.com/vcblog/2011/09/12/c11-features-in-visual-c-11/)
- [Visual Studio 2013 supports ```initializer_list```](https://msdn.microsoft.com/ja-jp/library/dn387584(v=vs.120).aspx)
- May be we should change the [if pragma here ](https://github.com/opencv/opencv/blob/dcb3c4ff1e70c7301a985cbf62da04d946b0a72b/modules/core/include/opencv2/core/cvdef.h#L378)

##### Steps to reproduce
 - build OpenCV master ( dcb3c4ff1e70c7301a985cbf62da04d946b0a72b ) on Visual Studio 2012

--- a/modules/core/include/opencv2/core/cvdef.h
+++ b/modules/core/include/opencv2/core/cvdef.h
@@ -375,7 +375,7 @@ Cv64suf;
 *                                    C++ 11                                              *
 \****************************************************************************************/
 #ifndef CV_CXX11
-#  if __cplusplus >= 201103L || (defined(_MSC_VER) && _MSC_VER >= 1600)
+#  if __cplusplus >= 201103L || (defined(_MSC_VER) && _MSC_VER >= 1800)
 #    define CV_CXX11 1
 #  endif
 #else

